Add getting started docs

sbar95 opened the pull request:
<!--
    Please read https://github.com/KohlsTechnology/eunomia/blob/master/.github/CONTRIBUTING.md before submitting
    your pull request. Please fill in each section below to help us better prioritize your pull request. Thanks!
-->

**Description**

<!-- Please provide a summary of the change here. -->
Add getting started guide for new users to kick off with Eunomia easier
and quicker. Provide comments to each command executed to make things
clear. Present an outline how Eunomia works.

Also adjust DEVELOPMENT.md and README.md to the getting started guide.

Add script to make it easier to develop eunomia locally. Previously the
user had to start minikube and set the env variables himself. Now this
is done for him in this script.

Include linux make in the deploy-to-local script - it shouldn't be decoupled
from this script. It must be run before building images. Now, there is no risk
that somebody will try to build eunomia-operator image without eunomia
binary being built.

Remove unnecessary commented-out lines from eunomia-hello-world-yaml
example.
<!-- Please link to all GitHub issue that this pull request implements(i.e. Fixes #123) -->
Fixes #197 

**Type of change**

<!-- Please delete options that are not relevant. -->

* New feature (non-breaking change which adds functionality)

**Checklist**

- [ ] Unit tests and e2e tests updated (not needed in this PR)
- [x] Documentation updated

Smiley73 commented on DEVELOPMENT.md:
````diff
@@ -1,61 +1,30 @@
-All the components can easily be installed via [Homebrew](https://brew.sh/) on a Mac:
+To achieve this, run the helper script:
 
 ```shell
````
What platforms are supported? This will obviously won't work on a Windows machine,

Smiley73 commented on scripts/run-eunomia-locally.sh:
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
```
Has this been tested on both Mac and Linux?

sbar95 replied:
Yes, this script and `run-eunomia-locally.sh`. They both work

Smiley73 commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,416 @@
+
+In this document you will find a step-by-step guide on how to get Eunomia running in a local Minikube cluster.
+
+The resources Eunomia, the GitOps controller, is watching and how it is done will be discussed in general.
```
Rephrase. Not sure what you're trying to say here.

MeganMoran replied:
+1

Smiley73 commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,416 @@
+- Access to a Kubernetes cluster
+  - [Minikube](https://kubernetes.io/docs/setup/minikube/) (for local development and testing)
+    - [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+  - [Minishift](https://www.okd.io/minishift/) (for local development and testing, optional)
```
The shell script you're providing doesn't work with minishift. So this might get confusing.

Smiley73 commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,416 @@
+
+### Checkout your fork
+
+Your own remote fork will be needed later to present how Eunomia adresses changes pushed to a repo being watched.
```
Rephrase, not a good sentence.

Smiley73 commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,416 @@
+
+Your own remote fork will be needed later to present how Eunomia adresses changes pushed to a repo being watched.
+
+To check out this repository:
```
Which repo? I assume you're implying the one that's currently looked at?

Smiley73 commented on GETTING_STARTED.md:
````diff
@@ -0,0 +1,416 @@
+2. Create the directories and clone your fork
+
+```
+mkdir -p $GOPATH/src/github.com/KohlsTechnology
````
GOPATH is probably not set. That is usually a separate step if I remember correctly.

Smiley73 commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,416 @@
+
+### Start a local Minikube cluster
+
+Minikube is a minimal kubernetes cluster run in a virtual machine in Virtualbox.
```
Minikube needs to be configured for virtualbox. We should have those instructions.
This is also great for a Mac, what about other platforms?

sbar95 replied:
Minikube by default automatically selects the virtualbox driver (I missed virtualbox in the prerequisites :/ ). But I agree - better add an explicit instruction.

The instructions for Mac already were in DEVELOPMENT.md, I just updated them to be sure that everything works and moved it to GETTING_STARTED.md. I'll add a general setup for Linux linking to vendor instructions - I could add code snippets for yum, apt, dnf, or pacman but it would be too much testing

MeganMoran commented on DEVELOPMENT.md:
````diff
@@ -1,61 +1,30 @@
 ```
 
-## Running Locally for Development Purposes
+If you want Eunomia to watch a particular namespace, just pass its name as a parameter to the above script. Running the script without the parameter defaults to Eunomia watching all namespaces.
````
What is the parameter name to be passed in?

MeganMoran commented on GETTING_STARTED.md:
````diff
@@ -0,0 +1,416 @@
+### Apply GitOpsConfig watching your remote branch
+
+```shell
+cat <<EOF | kubectl apply -n eunomia-hello-world-yaml-demo -f -
````
Adding the cat here just seems to complicate the instructions

sbar95 replied:
I intended this to be a code snippet for the user to just copy and paste to his own terminal. I also wanted the manifest to be visible in the GETTING_STARTED in order to have everything in one place.
Loading the manifest from stdin seemed most suitable for this example: https://kubernetes.io/docs/reference/kubectl/cheatsheet/#creating-objects

MeganMoran commented on README.md:
```diff
@@ -8,6 +8,10 @@
+
 ## Who is Eunomia
 
 According to Wikipedia:
```
Line 52, the first reference of "CR" should have the full work after i.e. "CR (Custom Resource)". Same with line 58 first reference of "CRD" should have definition (i.e. "CRD (Custom Resource Definition)")

MeganMoran commented on README.md:
```diff
@@ -8,6 +8,10 @@
+## Getting started
+
+If you want to deploy Eunomia from scratch to a local Minikube cluster, begin with the [Getting started](./GETTING_STARTED.md) document. It will guide your through the setup step-by-step.
+
```
Line 85 -   "ServiceAccountRef: "  should be lowercase based on code

MeganMoran commented on README.md:
```diff
@@ -8,6 +8,10 @@
+## Getting started
+
+If you want to deploy Eunomia from scratch to a local Minikube cluster, begin with the [Getting started](./GETTING_STARTED.md) document. It will guide your through the setup step-by-step.
+
```
![Gitops-hierarchy.png](./media/Gitops-hierarchy.png) - "Seeding CR team A" is referenced in 2 boxes, should the second be "Seeding CR team B"?

sbar95 replied:
@seanmalloy Do you have the original source of the image? I'd rather not edit the png

vinny-sabatini replied:
I believe @Smiley73 may have originally created that...but I'm not positive

MeganMoran commented on README.md:
```diff
@@ -8,6 +8,10 @@
+## Getting started
+
+If you want to deploy Eunomia from scratch to a local Minikube cluster, begin with the [Getting started](./GETTING_STARTED.md) document. It will guide your through the setup step-by-step.
+
```
The hierarchy section mentions "You can also use the relative path `./`, which means it'll also load the variables defined in `contextDir` directly (same folder that as `hierarchy.lst`)." How does this fit into priority? Would it be first or last priority when compared against the hierarchy.lst?

sbar95 replied:
It depends where you put it in the sequence - in [this example](https://github.com/KohlsTechnology/eunomia/blob/master/examples/hello-world-hierarchy/demo/hierarchy.lst) `./` has the highest priority.
But it's a good idea to clarify it

MeganMoran commented on README.md:
```diff
@@ -8,6 +8,10 @@
 [![Go Report Card](https://goreportcard.com/badge/github.com/KohlsTechnology/eunomia)](https://goreportcard.com/report/github.com/KohlsTechnology/eunomia)
 [![codecov](https://codecov.io/gh/KohlsTechnology/eunomia/branch/master/graph/badge.svg)](https://codecov.io/gh/KohlsTechnology/eunomia)
 
+## Getting started
```
In "Template Engine" section there is no info on where CA_BUNDLE and SERVICE_CA_BUNDLE should come from

sbar95 replied:
They are being set in the [discoverEnvironment.sh](https://github.com/KohlsTechnology/eunomia/blob/master/template-processors/base/bin/discoverEnvironment.sh#L30-L34) script depending on the target environment (k8s or openshift).

It's discussed here:
- https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod
- https://docs.openshift.com/container-platform/3.10/dev_guide/secrets.html#service-serving-certificate-secrets

I'll link it

akavel commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,415 @@
+  - [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+  - [Minikube](https://kubernetes.io/docs/setup/minikube/) (for local development and testing)
+
+You will also have to [set up the GOPATH](https://github.com/golang/go/wiki/SettingGOPATH) env variable (the instructions are for Go 1.13 but it's the same for Go 1.12).
```
I am not sure we need this text in parenthesis?

sbar95 replied:
Agree, for some reason I thought it's for Go 1.13 while it's only one of the options described there

akavel commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,415 @@
+
+You will also have to [set up the GOPATH](https://github.com/golang/go/wiki/SettingGOPATH) env variable (the instructions are for Go 1.13 but it's the same for Go 1.12).
+
+### Installing on a Mac
```
This seems to suggest to me like if eunomia could be only used on a Mac; should we also maybe add some info about Linux? if only to say "should work but not tested"? Though I had an impression that some people in our team do use Linux, maybe you could try asking around whether it's true and consulting with them if possible?

sbar95 replied:
I set Eunomia up on Linux some time ago, now I'm doing it again to make sure it still works. I'm also adding a general setup with links to vendor pages with instructions on how to install the prerequisites

akavel commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,415 @@
+
+### Checkout your fork
+
+You wil need your own Eunomia fork later in the second example. When pushing changes to your fork, you will observe Eunomia make adjustments based on your changed GitOpsConfig.
```
1. `wil` &rarr; `will`
 2. Could you explicitly use the name of the example you're referring to instead of "the second example"? currently I'm not sure which one exactly is this about, where should I start counting.
 3. I'm kinda uneasy with asking people to fork whole eunomia repo. If we only need this for people to easily modify examples, then I don't like the first sentence here.  To me, it makes it sound kinda like a fork of *whole eunomia repo* is *necessary* for even *getting started* with using eunomia. 
I think I'd prefer it rephrased to make it clear eunomia can work with any custom scratch-built repo with kubernetes/helm yaml files, but just to make instructions shorter and easier to maintain for us, we'll be asking user to fork the eunomia repos, purely for the examples/ directory (if that's what will be used)

akavel commented on GETTING_STARTED.md:
````diff
@@ -0,0 +1,415 @@
+```
+mkdir -p ${GOPATH}/src/github.com/KohlsTechnology
+cd ${GOPATH}/src/github.com/KohlsTechnology
+git clone https://github.com/<YOUR FORK>/eunomia.git
````
I suggest replacing space in `YOUR FORK` with underscore `_`; currently it makes it look like redirection operators, some users may try running this command verbatim and have weird results

akavel commented on GETTING_STARTED.md:
````diff
@@ -0,0 +1,415 @@
+```
+
+From now on your local Kubernetes client (kubectl) is configured to use your just started Minikube cluster.
+
````
Maybe add a link to minikube guide, to help users debug any problems with the setup, or otherwise make sure it's running correctly?

sbar95 replied:
Both on Linux and Mac it worked without any problems, but it's a good idea to add a link to some guide just in case

akavel commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,415 @@
+
+### Deploy Eunomia to your Minikube cluster
+
+It'll be done with the help of [Helm](https://helm.sh/). If you use a Mac, you can easily install it via Homebrew.
```
Shouldn't helm be part of prerequisites then?

sbar95 replied:
good point, I was writing the instructions on the fly

akavel commented on GETTING_STARTED.md:
````diff
@@ -0,0 +1,415 @@
+To achieve this run the helper script (assuming you are in your local Eunomia repository dir):
+
+```shell
+./scripts/deploy-to-local.sh minikube
````
Why not using official `latest` image of eunomia? This command is needed during development, or for bleeding-edge dev versions of eunomia; don't we want users to only use the latest full release of eunomia? who's the target of this getting started doc?

akavel commented on GETTING_STARTED.md:
````diff
@@ -0,0 +1,415 @@
+```shell
+helm template deploy/helm/eunomia-operator/ \
+  --set eunomia.operator.image.tag=dev \
+  --set eunomia.operator.image.pullPolicy=Never | kubectl apply -f -
````
Similar as above: don't we want users to stay with latest images, instead of dev? During our development, we specifically override tag=dev and pullPolicy=Never, so that we can leave latest and Always as defaults for end users. If we want to teach end users to also use dev and Never, there would be not much sense to keep latest and Always in the repo; but it's not yet clear to me why we would now want to teach users to use values other than latest and Always?

akavel commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,415 @@
+
+## Hello-world example
+
+We'll be using a very simple yaml files example which you can find [here](./examples/hello-world-yaml/README.md). Additionaly, we'll discuss every single step performed to make sure that everything regarding Eunomia operation is clear.
```
please make whole path visible instead of using `here` as link name, so that this guide stays usable when printed on paper or copy-pasted - e.g.:

    you can find in: [examples/hello-world-yaml/README.md](examples/hello-world-yaml/README.md).

akavel commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,415 @@
+|:---|:---|
+|  `kind`  |  set to `GitOpsConfig` - no Kubernetes application but Eunomia will be caring for it  |
+|  `templateSource` section  |  it defines the source of the templates  |
+|  - `uri`  |  repository where the templates are taken from (pay heed to the fact that this is [the official Eunomia repo](https://github.com/KohlsTechnology/eunomia), not your personal fork!)  |
```
[optional] how about a bit simpler wording like "pay attention" instead of "pay heed"?

akavel commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,415 @@
+|  `templateSource` section  |  it defines the source of the templates  |
+|  - `uri`  |  repository where the templates are taken from (pay heed to the fact that this is [the official Eunomia repo](https://github.com/KohlsTechnology/eunomia), not your personal fork!)  |
+|  - `ref`  |  branch on the remote repository  |
+|  - `contextDir`  |  directory where templates files will be taken from; in our example it's [this one](https://github.com/KohlsTechnology/eunomia/tree/master/examples/hello-world-yaml/template1) - you can find here a [single manifest file](https://github.com/KohlsTechnology/eunomia/blob/master/examples/hello-world-yaml/template1/hello-world.yaml) with a Deployment and a Service inside  |
```
similar as above: please don't use `this one` as link text, instead make the text understandable even if link targets were missing

akavel commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,415 @@
+|  `parameterSource` section  |  it defines the source of the parameters  |
+|  - `uri`  |  in our case there is no such key - so it is by default copied from the `templateSource` section  |
+|  - `ref`  |  branch on the remote repository  |
+|  - `contextDir`  |  directory where parameter files will be taken from; in our example it's [this one](https://github.com/KohlsTechnology/eunomia/tree/master/examples/hello-world-yaml/parameters) - as you can see it's rather empty, and that's because in our example we don't parametrize anything (everything is set to a fixed value in the templates directory already)  |
```
see l.156

sbar95 replied:
Marking for myself as not outdated

akavel commented on GETTING_STARTED.md:
````diff
@@ -0,0 +1,415 @@
+2. Create the directories and clone your fork:
+
+```
+mkdir -p ${GOPATH}/src/github.com/KohlsTechnology
````
starting with Go 1.12, I believe it's not required to clone into GOPATH; for what it's worth, I personally work and develop on eunomia cloned into a non-GOPATH directory

sbar95 replied:
I'll check it

sbar95 replied:
You're correct, it's not needed

sbar95 commented:
All, thank you for your feedback on this PR.

I have simplified the getting started instructions to include only a easy deployment via OperatorHub. I've also adjusted the DEVELOPMENT.md file.

As for now I've removed the example with setting up the GitHub webhook as it was rather a getting started for a developer, not an end user - I'll probably include it in the DEVELOPMENT.md in a future PR.

Please, run through my PR again to see if everything is acceptable now. If so, I'll squash the commits.

akavel commented on DEVELOPMENT.md:
```diff
@@ -2,15 +2,41 @@
-- Access to a Kubernetes cluster
-  - [Minikube](https://kubernetes.io/docs/setup/minikube/) (optional)
-  - [Minishift](https://www.okd.io/minishift/) (optional)
+- [Minishift](https://github.com/minishift/minishift/releases) (optional; a local OpenShift cluster)
```
What would you think about keeping the list as it was before (still mentioning minikube), but adding a link to Getting Started to follow for detailed guide? Also because IIUC, if using minishift, user wouldn't actually need minikube and the other stuff from Prerequisites, no?

sbar95 replied:
I'll rework it in such a way that all instructions will be in one place and I'll be only referencing them from everywhere

akavel replied:
I'm ok with some information being repeated/duplicated, I believe it can sometimes help readability. I see DEVELOPMENT.md vs. GETTING_STARTED.md as having very different target audience and goal. For example, I could imagine thinking of DEVELOPMENT as having comprehensive & exhaustive "RTFM-grade" info, while GETTING_STARTED being a single, focused, "simple happy path" tutorial/story, with some aspects repeated from DEVELOPMENT.

akavel commented on DEVELOPMENT.md:
```diff
@@ -2,15 +2,41 @@
+brew install go@1.12
+brew install docker
+brew cask install minishift  # optional
+brew instal openshift-cli  # optional
```
I am confused why this list mentions minishift, but doesn't mention minikube; I would expect to see explicit choice between minishift and minikube here.

akavel commented on DEVELOPMENT.md:
```diff
@@ -2,15 +2,41 @@
+|    |    |    |
+|:---|:---|:---|
+|  [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)  |  [Go](https://golang.org/doc/install#tarball)  |  [Docker](https://docs.docker.com/install/)  |
+|  [Minishift](https://github.com/minishift/minishift/releases) (optional)  |  [oc](https://docs.openshift.com/container-platform/3.7/cli_reference/get_started_cli.html#installing-the-cli) (optional)  |    |
```
Again, I'm missing info on minikube. 

Also, similar question like in Getting Started with regards to the table.

akavel commented on scripts/run-eunomia-locally.sh:
```diff
@@ -0,0 +1,39 @@
+export CRONJOB_TEMPLATE=./build/job-templates/cronjob.yaml
+# Setting WATCH_NAMESPACE to an empty string results in Eunomia watching
+# all namespaces (operator-sdk internal)
+export WATCH_NAMESPACE="$watch_ns"
```
does this need to be exported? it seems used on l.39, is it additionally loaded from env by operator-sdk?

sbar95 replied:
Good point, it's passed as `operator-sdk`'s parameter and then set inside as an env variable in a process run with `exec`. No need to export it in the script

akavel commented on DEVELOPMENT.md:
````diff
@@ -19,47 +45,36 @@ To check out this repository:
+Eunomia binary being run locally still uses [template-processor images](./README.md#Template-Engine) to manage templates and parameters. If you are tweaking the images (anything in `template-processors` directory), make sure to deploy them to your local Minikube Docker registry beforehand. The easiest way is to use the helper script:
+
+```shell
+./scripts/deploy-to-local.sh minikube
````
or minishift?

sbar95 replied:
In 9b07ac9feb5d95390d9e96b0f4520a55e4e9561a I added a note that running eunomia locally enforces the use of a Kubernetes cluster

vinny-sabatini commented on README.md:
```diff
@@ -8,6 +8,10 @@
+
+If you want to deploy Eunomia from scratch to a local Minikube cluster, begin with the [Getting started](./GETTING_STARTED.md) document. It will guide your through the setup step-by-step.
+
 ## Who is Eunomia
```
[Optional] I think we should move the `Who is Eunomia`, `What is GitOps`, and `Next Generation Change Management` sections to a different page, or at least later in the page. In my opinion the README.md should start off with what Eunomia is and what it does so the user knows as soon as they open the repo.

If we add the section for what is Eunomia/what does it do, I think it should be before the `Getting Started` section, because the user may not even know what they are starting.

This can also be done in a separate documentation cleanup effort, as I believe the issue you are currently tackling is simply a getting started guide.

akavel replied:
FWIW, personally, I think I'd prefer if we tried doing this in a separate PR, to keep focus. Both @sbar95's work and our reviews here are difficult enough IMO, I'm somewhat afraid of spreading our effort over multiple concerns

sbar95 replied:
You're right - but in this PR I'll just move the `Getting started` section after `Purpose`

vinny-sabatini replied:
I've created #289 to look into cleaning up the README in a future PR.

vinny-sabatini commented on README.md:
```diff
@@ -226,12 +230,12 @@ A base image is provided that can be inherited to simplify the process of adding
-    | `CA_BUNDLE`  | Path to the platform-level CA bundle  |
-    | `SERVICE_CA_BUNDLE`  | Path to the service-level CA bundle  |
+    | `CA_BUNDLE`  | Path to the [platform-level CA bundle](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod)  |
+    | `SERVICE_CA_BUNDLE`  | Path to the [service-level CA bundle](https://docs.openshift.com/container-platform/3.10/dev_guide/secrets.html#service-serving-certificate-secrets)  |
```
```suggestion
    | `SERVICE_CA_BUNDLE`  | Path to the [service-level CA bundle](https://docs.openshift.com/container-platform/3.11/dev_guide/secrets.html#service-serving-certificate-secrets)  |
```

sbar95 commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,215 @@
+
+|    |    |    |
+|:---|:---|:---|
+|  [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux)  |  [VirtualBox](https://www.virtualbox.org/wiki/Linux_Downloads)  |  [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/)  |
```
The preview in my local environment looked better than the version on github... I'll change it to something simpler

sbar95 commented on README.md:
````diff
@@ -144,7 +148,7 @@ Example `hierarchy.lst`:
 ```
 
-In this case it will load all yaml files from `../defaults`, then merge it with everything in `../marketing`, and lastly merges it with everything in `../development`. You can also use the relative path `./`, which means it'll also load the variables defined in `contextDir` directly (same folder that as `hierarchy.lst`).
+In this case it will load all yaml files from `../defaults`, then merge it with everything in `../marketing`, and lastly merges it with everything in `../development`. You can also use the relative path `./`, which means it'll also load the variables defined in `contextDir` directly (same folder that as `hierarchy.lst`). You can insert the ralative path wherever you want in the `hierarchy.lst` -  ...
````
I added this as further clarification in response to https://github.com/KohlsTechnology/eunomia/pull/281#discussion_r373695413. I wanted that there wouldn't be any doubts for anybody

akavel replied:
After seeing the question from Megan, I think if you changed *"You can insert the relative path..."* to *"You can insert `./`..."* I would be ok with this. Unless I also misunderstand something.

akavel commented on GETTING_STARTED.md:
````diff
@@ -0,0 +1,215 @@
+minikube start --vm-driver virtualbox
+```
+
+From now on your local Kubernetes client (kubectl) is configured to use your just started Minikube cluster. If any problems should occur, you might want to take a look at this simple Minikube guide: https://kubernetes.io/docs/setup/learning-environment/minikube/#quickstart
````
`at this simple [Minikube guide]` &rarr; `at the [Minikube quickstart guide]` ?

sbar95 commented on GETTING_STARTED.md:
````diff
@@ -0,0 +1,215 @@
+
+The one from [`examples/hello-world-yaml/cr/hello-world-cr1.yaml`](./examples/hello-world-yaml/cr/hello-world-cr1.yaml) will be applied:
+
+```yaml
````
I wanted it to be here, visible for the user without a need to go to other places, but you're right - we might change it at the source and forget to do it here

sbar95 commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,215 @@
+
+Things to pay attention to:
+
+|  field name  |  description  |
```
Here the table describes a specific CR - it's for the user to get familiar on a basic level. In README.md the description is more general and touches every possible field; here only the ones needed to understand how Eunomia operates in the example

sbar95 commented on GETTING_STARTED.md:
```diff
@@ -0,0 +1,215 @@
+# Getting started
+
+In this document you will find a step-by-step guide on how to get Eunomia running in a local Minikube cluster.
```
I was also thinking about writing an instruction to get Eunomia running in a Kubernetes cluster in general (Minikube as just an option) but for the sake of simplicity I decided to keep it as it is. If the user already has access to his own Kubernetes cluster configured, I think he'll know exactly what steps to skip to deploy Eunomia to his cluster

seanmalloy commented:
I think it looks good.

@akavel @vinny-sabatini @Smiley73 @MeganMoran thanks for your detailed reviews. Please review this pull request one more time and let @sbar95 know if any additional changes are need or approve the PR it is looks good now.

akavel commented on DEVELOPMENT.md:
```diff
@@ -19,47 +92,40 @@ To check out this repository:
-export GO111MODULE=on
-go mod vendor
-operator-sdk up local --namespace="${WATCH_NAMESPACE}"
+The above script uses `operator-sdk up local` command to run Eunomia operator binary locally. **It enforces the use of a Kubernetes cluster** - Eunomia will be watching namespace(s) in a Kubernetes cluster. Using an OpenShift cluster is unconfigurable, unfortunately.
```
[optional] I don't really understand what the "unconfigurable" sentence means; what would you think about removing it?

```suggestion
The above script uses `operator-sdk up local` command to run Eunomia operator binary locally. **It enforces the use of a Kubernetes cluster** - Eunomia will be watching namespace(s) in a Kubernetes cluster.
```
